middleware/users.js

```diff
@@ -0,0 +1,57 @@
+const errors = require("../configuration/errors");
+const { updateUserDataSchema } = require("./../utils/validation_schemas/update-user-data")
+
+const validatePageAndPageSize = async function (req, res, next) {
+
+    if (!req.query.page) {
+        req.query.page = 1
+    }
+
+    if (req.query.page < 1) {
+        return res.status(400).json({
+            error: errors.PAGE_NUMBER_LESS_THAN_1
+        })
+    }
+
+    if (!req.query.size) {
+        req.query.size = 0
+    }
+
+    //if page size is zero, mongoose will get entire collection
+    if (req.query.size < 0) {
```

Comment:
I think we shouldn't get all collection in any case. What if we have 10k rows. Our backend will fall down.